### PR TITLE
[date-time-picker]Fix for space and label issues

### PR DESCRIPTION
**Pull request by ashishkcerner**
### Summary
 Fix the spacing between datetimepicker and help/error text and label announcement in screen readers

**What was changed:**

**Why it was changed:**

### Testing

This change was tested using:

- [x] WDIO
- [x] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [x] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-8025 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -80,7 +80,7 @@
     "start-static": "npm run compile:prod && terra express-server --site build",
     "test": "npm run lint && npm run jest && npm run wdio",
     "test:docker": "npm run lint && npm run jest && npm run wdio:docker",
-    "wdio": "terra wdio --themes terra-default-theme clinical-lowlight-theme orion-fusion-theme",
+    "wdio": "terra wdio --themes terra-default-theme clinical-lowlight-theme orion-fusion-theme --spec packages/terra-date-time-picker/tests/wdio/date-time-picker-mobile-spec.js",
```
Comment (saket2403) [inline package.json:83]: Revert this change

Comment (ashishkcerner) [reply package.json]: Thanks done https://github.com/cerner/terra-framework/pull/1969/commits/5723e357fafd47dd891a3ecc9b57179b20709a08

File: packages/terra-date-picker/src/DateInput.jsx [hunk 2]
```diff
@@ -733,7 +733,7 @@ const DatePickerInput = (props) => {
       size="2"
       pattern="\d*"
       aria-required={required}
-      aria-label={intl.formatMessage({ id: 'Terra.datePicker.dayLabel' })}
+      aria-label={ariaLabel ? `${ariaLabel} ${intl.formatMessage({ id: 'Terra.datePicker.dayLabel' })} ` : intl.formatMessage({ id: 'Terra.datePicker.dayLabel' })}
```
Comment (supreethmr) [inline packages/terra-date-picker/src/DateInput.jsx:736]: So that screen reader gives pause after reading aria-label value
```suggestion
      aria-label={ariaLabel ? `${ariaLabel}, ${intl.formatMessage({ id: 'Terra.datePicker.dayLabel' })} ` : intl.formatMessage({ id: 'Terra.datePicker.dayLabel' })}
```

**Review by supreethmr — CHANGES_REQUESTED** [review]
We need to review this again, as screen readers were not able to communicate the label value as expected.

File: packages/terra-date-picker/src/DateInput.jsx [hunk 3]
```diff
@@ -734,7 +734,7 @@ const DatePickerInput = (props) => {
       pattern="\d*"
       aria-required={required}
       aria-label={intl.formatMessage({ id: 'Terra.datePicker.dayLabel' })}
-      aria-describedby={dateFormatOrder === DateUtil.dateOrder.DMY ? `${nameLabelId} ${ariaDescriptionIds}` : ariaDescriptionIds}
+      aria-describedby={`${nameLabelId} ${ariaDescriptionIds}`}
```
Comment (supreethmr) [inline packages/terra-date-picker/src/DateInput.jsx:737]:
```suggestion
      aria-describedby={`${nameLabelId}, ${ariaDescriptionIds}`}
```

File: packages/terra-date-picker/src/DateInput.jsx [hunk 4]
```diff
@@ -765,7 +765,7 @@ const DatePickerInput = (props) => {
       pattern="\d*"
       aria-required={required}
       aria-label={intl.formatMessage({ id: 'Terra.datePicker.monthLabel' })}
-      aria-describedby={dateFormatOrder === DateUtil.dateOrder.MDY ? `${nameLabelId} ${ariaDescriptionIds}` : ariaDescriptionIds}
+      aria-describedby={`${nameLabelId} ${ariaDescriptionIds}`}
```
Comment (supreethmr) [inline packages/terra-date-picker/src/DateInput.jsx:768]:
```suggestion
      aria-describedby={`${nameLabelId}, ${ariaDescriptionIds}`}
```

File: packages/terra-date-picker/src/DateInput.jsx [hunk 5]
```diff
@@ -796,7 +796,7 @@ const DatePickerInput = (props) => {
       pattern="\d*"
       aria-required={required}
       aria-label={intl.formatMessage({ id: 'Terra.datePicker.yearLabel' })}
-      aria-describedby={dateFormatOrder === DateUtil.dateOrder.YMD ? `${nameLabelId} ${ariaDescriptionIds}` : ariaDescriptionIds}
+      aria-describedby={`${nameLabelId} ${ariaDescriptionIds}`}
```
Comment (supreethmr) [inline packages/terra-date-picker/src/DateInput.jsx:799]:
```suggestion
      aria-describedby={`${nameLabelId}, ${ariaDescriptionIds}`}
```

File: packages/terra-date-time-picker/src/DateTimePicker.jsx [hunk 6]
```diff
@@ -59,6 +59,10 @@ const propTypes = {
   * Whether the selected meridiem displays as Invalid. Use when value does not meet validation pattern.
   */
   isInvalidMeridiem: PropTypes.bool,
+  /**
+   * The label of the form control children.
+   */
+  label: PropTypes.node,
```
Comment (supreethmr) [inline packages/terra-date-time-picker/src/DateTimePicker.jsx:65]: should this be renamed as ariaLabel since it is not used for visual display

File: packages/terra-time-input/src/TimeInput.jsx [hunk 7]
```diff
@@ -1106,7 +1106,7 @@ class TimeInput extends React.Component {
             {...inputAttributes}
             {...minuteAttributes}
             refCallback={(inputRef) => { this.minuteInput = inputRef; }}
-            label={intl.formatMessage({ id: 'Terra.timeInput.minutes' })}
+            label={intl.formatMessage({ id: 'Terra.timeInput.minutes' }, { a11yLabel: this.a11yLabel })}
```
Comment (supreethmr) [inline packages/terra-time-input/src/TimeInput.jsx:1109]: why do we need to send this as object value ..??

File: packages/terra-date-picker/src/DateInput.jsx (hunk 3 above)
Comment (ashishkcerner) [reply packages/terra-date-picker/src/DateInput.jsx:737]: The utilization of commas in this context may lead to suboptimal behavior with assistive technologies -as per the MDN websites

File: packages/terra-date-picker/src/DateInput.jsx (hunk 4 above)
Comment (ashishkcerner) [reply packages/terra-date-picker/src/DateInput.jsx:768]: The utilization of commas in this context may lead to suboptimal behavior with assistive technologies -as per the MDN websites

File: packages/terra-date-picker/src/DateInput.jsx (hunk 5 above)
Comment (ashishkcerner) [reply packages/terra-date-picker/src/DateInput.jsx:799]: The utilization of commas in this context may lead to suboptimal behavior with assistive technologies -as per the MDN websites

File: packages/terra-time-input/src/TimeInput.jsx (hunk 7 above)
Comment (ashishkcerner) [reply packages/terra-time-input/src/TimeInput.jsx:1109]: In this case i have followed the previous code standard as it is being used for Hours field

**Comment by supreethmr** [comment]
@ashishkcerner  I would suggest to update this PR only with fix for space issue ( OR create a new PR only with space fix)  and label issue can be re-validated after some of the existing PR's are merged. I see this PR also addressing issue of label ( error message ) so it's better re-validate the label reading issue after the PR https://github.com/cerner/terra-framework/pull/1975. is merged

**Comment by ashishkcerner** [comment]
Have raised a new PR excluding label screenreader reading issue .https://github.com/cerner/terra-framework/pull/1997
